docs/_docs/sql/java.adoc

```diff
@@ -0,0 +1,68 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+= Java SQL API
+
+In your Java projects, you can use the Java SQL API to execute SQL statements and getting results. All operations are executed as a part of sql session. You can create a session with default parameters by using a `sql.createSession()` method, or use an `sql.sessionBuilder` to configure it for your environment. The `execute` method accepts a query String, or a Statement object, that can be created by using `sql.createStatement()` and `sql.statementBuilder()` methods.
+
+== Creating Tables
+
+Here is an example of how you can create a new table on a cluster:
+
+[source, java]
+----
+IgniteSql sql = CLUSTER_NODES.get(0).sql();
+Session ses = sql.createSession();
+ResultSet synchRes = ses.execute(
+    null,
+    `CREATE TABLE SYNCH(ID INT PRIMARY KEY, VAL0 INT)`
+);
+
+AsyncResultSet asyncRes = fut.get();
+
+ses.closeAsync().get();
+----
+
+
+== Filling Tables
+
+With Apache Ignite 3, you can fill the table by adding rows one by one, or in a batch. In both cases, you create an `INSERT` statement, and then exeute it:
+
+
+[source, java]
+----
+long rowsAdded = Arrays.stream(ses.executeBatch(null,
+    "INSERT INTO ACCOUNTS (ACCOUNT_ID, CITY_ID, FIRST_NAME, LAST_NAME, BALANCE) values (?, ?, ?, ?, ?)",
+    BatchedArguments.of(1, 1, "John", "Doe", 1000.0d)
+        .add(2, 1, "Jane", "Roe", 2000.0d)
+        .add(3, 1, "Mary", "Major", 1500.0d)
+        .add(4, 1, "Richard", "Miles", 1450.0d)))
+        .asLongStream().sum();
+----
+
+== Getting Data From Tables
+
+To get data from a table, execute a `SELECT` statement to get a set of results. You can then iterate through results to get data:
```

Comment:
It worth to mention that
ResultSet may hold server-side cursor open due to lazy query execution and must be closed manually or via using try-with-resources block.